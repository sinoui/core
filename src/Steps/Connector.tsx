import React from 'react';
import styled from 'styled-components';
import Divider from '../Divider';

export interface Props {
  complete?: boolean;
}

const ConnectorLayout = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const StyledDivider = styled(Divider)<{ complete?: boolean }>`
  flex: 1;
  background: ${({ complete, theme }) =>
    complete ? theme.palette.primary.main : theme.palette.text.disabled};
`;

export default function Connector(props: Props) {
  const { complete } = props;
  return (
    <ConnectorLayout>
      <StyledDivider complete={complete} />
    </ConnectorLayout>
  );
}
