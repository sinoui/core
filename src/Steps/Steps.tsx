import React from 'react';
import styled, { css } from 'styled-components';
import classNames from 'classnames';

export interface StepsProps {
  /**
   * 当前处于活动状态的步骤条
   */
  current: number;
  /**
   * 步骤条方向
   */
  direction?: 'horizontal' | 'vertical';
  /**
   * 是否为非线性，默认为false
   */
  nonLinear?: boolean;
  /**
   * 子元素
   */
  children?: React.ReactNode;
  /**
   * 标签位置
   */
  labelPlacement?: 'horizontal' | 'vertical';
  /**
   * 连接器
   */
  connector?: React.ReactNode;
  /**
   * 自定义图标
   */
  icon?: React.ReactNode;
  /**
   * 点击时的回调函数
   */
  onChange?: (current: number) => void;
}

const verticalStyle = css`
  align-items: flex-start;
`;

const StepsLayout = styled.div<{ labelPlacement?: 'horizontal' | 'vertical' }>`
  display: flex;
  align-items: center;
  ${({ labelPlacement }) => labelPlacement === 'vertical' && verticalStyle}

  .sinoui-step-connector--active {
    color: ${({ theme }) => theme.palette.primary.main};

    > hr {
      background: ${({ theme }) => theme.palette.primary.main};
    }
  }
`;

export default function Steps(props: StepsProps) {
  const {
    direction,
    children,
    current,
    nonLinear,
    labelPlacement,
    connector,
    icon,
    onChange,
    ...other
  } = props;
  const childrenArray = React.Children.toArray(children);
  const steps = childrenArray.map((step: any, index) => {
    const state = {
      index,
      active: false,
      completed: false,
      disabled: false,
    };

    if (current === index) {
      state.active = true;
    } else if (current > index) {
      state.completed = true;
    } else if (current < index) {
      state.disabled = true;
    }

    return React.cloneElement(step, {
      last: index + 1 === childrenArray.length,
      connector,
      labelPlacement,
      icon,
      onChange,
      ...state,
      ...step.props,
    });
  });

  return (
    <StepsLayout
      {...other}
      labelPlacement={labelPlacement}
      className={classNames('sinoui-steps', {
        'sinoui-steps--horizontal': direction === 'horizontal',
        'sinoui-steps--vertical': direction === 'vertical',
      })}
    >
      {steps}
    </StepsLayout>
  );
}
