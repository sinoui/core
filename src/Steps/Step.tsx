import React from 'react';
import styled, { css } from 'styled-components';
import Check from '@sinoui/icons/Check';
import { useRipple } from '@sinoui/ripple';
import classNames from 'classnames';
import Connector from './Connector';

const verticalLayoutCss = css`
  flex: 1;
  position: relative;

  .sinoui-step-connector {
    position: absolute;
    top: 12px;
    left: calc(50% + 20px);
    right: calc(-50% + 20px);
    margin-top: 0px;
  }

  .sinoui-step-connector--active {
    color: ${({ theme }) => theme.palette.primary.main};

    > hr {
      background: ${({ theme }) => theme.palette.primary.main};
    }
  }
`;

const StepLayout = styled.div<{
  $isVertical?: boolean;
  $isClickable?: boolean;
}>`
  padding: 0 16px;
  box-sizing: border-box;
  cursor: ${({ $isClickable }) => ($isClickable ? 'pointer' : 'default')};
  ${({ $isVertical }) => $isVertical && verticalLayoutCss};

  .sinoui-step-container--vertical {
    flex-direction: column;
    align-items: center;

    .sinoui-step-number-icon {
      margin-right: 0;
    }

    .sinoui-step-content {
      margin-top: 16px;
      text-align: center;
    }
  }
`;

const Title = styled.span<{ $active?: boolean }>`
  color: ${({ theme, $active }) =>
    $active ? theme.palette.text.primary : theme.palette.text.disabled};
  font-size: 14px;
  font-weight: ${({ theme, $active }) =>
    $active
      ? theme.typography.fontWeightMedium
      : theme.typography.fontWeightRegular};
`;

const Description = styled.div<{ $active?: boolean }>`
  color: ${({ theme, $active }) =>
    $active ? theme.palette.text.primary : theme.palette.text.disabled};
`;

const StepContent = styled.div<{ $isVertical?: boolean }>`
  display: inline-block;
  vertical-align: top;
`;

const NumberIcon = styled.div<{ $active?: boolean }>`
  width: 24px;
  height: 24px;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  margin-right: 8px;
  font-size: 12px;
  line-height: 24px;
  color: ${({ theme }) => theme.palette.common.white};
  background: ${({ theme, $active }) =>
    $active ? theme.palette.primary.main : theme.palette.text.disabled};
`;

const IconWrapper = styled.span<{ $active?: boolean }>`
  margin-right: 8px;
  color: ${({ theme, $active }) =>
    $active ? theme.palette.primary.main : theme.palette.text.disabled};
`;

const StepContainer = styled.div`
  display: flex;
  align-items: flex-start;
`;

const SubTitle = styled.div`
  display: inline;
  margin-left: 8px;
  color: ${({ theme }) => theme.palette.text.disabled};
`;

export interface StepProps {
  /**
   * 自定义图标
   */
  icon?: React.ReactNode;
  /**
   * 自定义连接器
   */
  connector?: React.ReactNode;
  /**
   * 标题
   */
  title: string;
  /**
   * 编号
   */
  index?: number;
  /**
   * 不可用状态
   */
  disabled?: boolean;
  /**
   * 是否是最后一个
   */
  last?: boolean;
  /**
   * 标签位置
   */
  labelPlacement?: 'horizontal' | 'vertical';
  /**
   * 点击时的回调函数
   */
  onChange?: (current: number) => void;
  /**
   * 步骤条方向
   */
  direction?: 'horizontal' | 'vertical';
  /**
   * 状态
   */
  status?: 'wait' | 'active' | 'completed' | 'error';
  /**
   * 副标题
   */
  subTitle?: React.ReactNode;
  /**
   * 描述文字（内容）
   */
  description?: React.ReactNode;
}

export default function Step(props: StepProps) {
  const {
    title,
    index = 0,
    last,
    connector,
    labelPlacement,
    icon,
    onChange,
    direction,
    status,
    subTitle,
    description,
  } = props;

  const ref = onChange ? useRipple<HTMLDivElement>() : null;

  const handleClick = () => {
    if (!onChange) {
      return;
    }
    onChange(index);
  };
  return (
    <>
      <StepLayout
        className="sinoui-step"
        $isVertical={labelPlacement === 'vertical'}
        onClick={handleClick}
        $isClickable={!!onChange}
        ref={ref}
      >
        {labelPlacement === 'vertical' && direction !== 'vertical' && !last && (
          <Connector
            className={classNames('sinoui-step-connector', {
              'sinoui-step-connector--active': status === 'completed',
            })}
          >
            {connector}
          </Connector>
        )}
        <StepContainer
          className={classNames('sinoui-step-container', {
            'sinoui-step-container--vertical': labelPlacement === 'vertical',
          })}
        >
          {icon ? (
            <IconWrapper
              $active={status === 'active' || status === 'completed'}
            >
              {icon}
            </IconWrapper>
          ) : (
            <NumberIcon
              $active={status === 'active' || status === 'completed'}
              className="sinoui-step-number-icon"
            >
              {status === 'completed' ? <Check size={18} /> : index + 1}
            </NumberIcon>
          )}
          <StepContent className="sinoui-step-content">
            <Title
              $active={status === 'active' || status === 'completed'}
              className="sinoui-step-title"
            >
              {title}
              {subTitle && (
                <SubTitle className="sinoui-step-subtitle">{subTitle}</SubTitle>
              )}
            </Title>
            {description && (
              <Description
                className="sinoui-step-description"
                $active={status === 'active' || status === 'completed'}
              >
                {description}
              </Description>
            )}
          </StepContent>
        </StepContainer>
      </StepLayout>
      {!last &&
        (labelPlacement !== 'vertical' ||
          (labelPlacement === 'vertical' && direction === 'vertical')) && (
          <Connector
            className={classNames('sinoui-step-connector', {
              'sinoui-step-connector--active': status === 'completed',
            })}
          >
            {connector}
          </Connector>
        )}
    </>
  );
}
