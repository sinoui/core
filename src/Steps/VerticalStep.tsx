/* eslint-disable no-nested-ternary */
import React from 'react';
import styled from 'styled-components';
import { useRipple } from '@sinoui/ripple';
import classNames from 'classnames';
import CheckCircle from './CheckCircleIcon';
import Error from './ErrorIcon';
import Connector from './Connector';

const StepLayout = styled.div<{
  $isVertical?: boolean;
  $isClickable?: boolean;
}>`
  padding: 0 16px;
  box-sizing: border-box;
  cursor: ${({ $isClickable }) => ($isClickable ? 'pointer' : 'default')};
`;

const Title = styled.span<{ $active?: boolean; $error?: boolean }>`
  color: ${({ theme, $active, $error }) =>
    $active
      ? theme.palette.text.primary
      : $error
      ? theme.palette.error.main
      : theme.palette.text.disabled};
  font-size: 14px;
  line-height: 24px;
  font-weight: ${({ theme, $active }) =>
    $active
      ? theme.typography.fontWeightMedium
      : theme.typography.fontWeightRegular};
`;

const Description = styled.div<{ $active?: boolean }>`
  color: ${({ theme, $active }) =>
    $active ? theme.palette.text.secondary : theme.palette.text.disabled};
  padding-bottom: 16px;
`;

const StepContent = styled.div<{ $isVertical?: boolean }>`
  display: inline-block;
  vertical-align: top;
`;

const NumberIcon = styled.div<{ $active?: boolean }>`
  width: 24px;
  height: 24px;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  font-size: 12px;
  color: ${({ theme }) => theme.palette.common.white};
  background: ${({ theme, $active }) =>
    $active ? theme.palette.primary.main : theme.palette.text.disabled};
`;

const IconWrapper = styled.span<{ $active?: boolean; $error?: boolean }>`
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  color: ${({ theme, $active }) =>
    $active ? theme.palette.primary.main : theme.palette.text.disabled};
  ${({ theme, $error }) => $error && `color:${theme.palette.error.main}`};
`;

const StepContainer = styled.div`
  display: grid;
  display: -ms-grid;
  -ms-grid-columns: 48px auto;
  -ms-grid-rows: auto auto;
  grid-template-columns: 48px auto;
  grid-template-rows: auto auto;

  .sinoui-step-icon {
    -ms-grid-column: 1;
    -ms-grid-row: 1;
  }

  .sinoui-step-content {
    -ms-grid-column: 2;
    -ms-grid-row: 1;
  }

  .sinoui-step-connector {
    -ms-grid-column: 1;
    -ms-grid-row: 2;
  }

  .sinoui-step-description {
    -ms-grid-column: 2;
    -ms-grid-row: 2;
  }

  hr {
    height: calc(100% - 8px);
  }
`;

const SubTitle = styled.div`
  display: inline;
  margin-left: 8px;
  color: ${({ theme }) => theme.palette.text.disabled};
`;

export interface VerticalStepProps {
  /**
   * 自定义图标
   */
  icon?: React.ReactNode;
  /**
   * 自定义连接器
   */
  connector?: React.ReactNode;
  /**
   * 标题
   */
  title: string;
  /**
   * 编号
   */
  index?: number;
  /**
   * 不可用状态
   */
  disabled?: boolean;
  /**
   * 是否是最后一个
   */
  last?: boolean;
  /**
   * 点击时的回调函数
   */
  onChange?: (current: number) => void;
  /**
   * 状态
   */
  status?: 'wait' | 'active' | 'completed' | 'error';
  /**
   * 副标题
   */
  subTitle?: React.ReactNode;
  /**
   * 描述文字（内容）
   */
  description?: React.ReactNode;
}

/**
 * 垂直方向步进器
 * @param props
 */
export default function VerticalStep(props: VerticalStepProps) {
  const {
    title,
    index = 0,
    last,
    connector,
    icon,
    onChange,
    status,
    subTitle,
    description,
  } = props;

  const ref = useRipple<HTMLDivElement>({
    disabled: !onChange,
  });

  const renderIcon = () => {
    if (icon) {
      if (typeof icon === 'function') {
        return icon(status);
      }
      return (
        <IconWrapper
          $active={status === 'active' || status === 'completed'}
          $error={status === 'error'}
          className="sinoui-step-icon"
        >
          {icon}
        </IconWrapper>
      );
    }
    if (status === 'completed') {
      return (
        <IconWrapper $active className="sinoui-step-icon">
          <CheckCircle />
        </IconWrapper>
      );
    }
    if (status === 'error') {
      return (
        <IconWrapper className="sinoui-step-icon" $error>
          <Error />
        </IconWrapper>
      );
    }

    return (
      <IconWrapper className="sinoui-step-icon">
        <NumberIcon $active={status === 'active'} className="sinoui-step-icon">
          {index + 1}
        </NumberIcon>
      </IconWrapper>
    );
  };

  const handleClick = () => {
    if (!onChange) {
      return;
    }
    onChange(index);
  };
  return (
    <>
      <StepLayout
        className="sinoui-step"
        onClick={handleClick}
        $isClickable={!!onChange}
        ref={ref}
      >
        <StepContainer className="sinoui-step-container">
          {renderIcon()}
          <StepContent className="sinoui-step-content">
            <Title
              $active={status === 'active' || status === 'completed'}
              $error={status === 'error'}
              className="sinoui-step-title"
            >
              {title}
              {subTitle && (
                <SubTitle className="sinoui-step-subtitle">{subTitle}</SubTitle>
              )}
            </Title>
          </StepContent>
          {!last ? (
            <Connector
              className={classNames('sinoui-step-connector', {
                'sinoui-step-connector--active': status === 'completed',
              })}
            >
              {connector}
            </Connector>
          ) : (
            <div className="sinoui-step-connector" />
          )}
          {description && (
            <Description
              className="sinoui-step-description"
              $active={status === 'active' || status === 'completed'}
            >
              {description}
            </Description>
          )}
        </StepContainer>
      </StepLayout>
    </>
  );
}
