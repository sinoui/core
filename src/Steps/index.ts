import Steps from './Steps';
import Step from './Step';

export default Steps;

export { Step };
