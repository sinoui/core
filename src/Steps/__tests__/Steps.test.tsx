/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import '@testing-library/jest-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Steps from '../Steps';
import Step from '../Step';

afterEach(cleanup);

it('基础渲染', () => {
  const tree = renderer.create(
    <ThemeProvider theme={defaultTheme}>
      <Steps current={0}>
        <Step title="第一步" />
        <Step title="第二步" />
        <Step title="第三步" />
      </Steps>
    </ThemeProvider>,
  );

  expect(tree).toMatchSnapshot();
});

it('正确渲染--标签垂直布局', () => {
  const tree = renderer.create(
    <ThemeProvider theme={defaultTheme}>
      <Steps current={0} labelPlacement="vertical">
        <Step title="第一步" />
        <Step title="第二步" />
        <Step title="第三步" />
      </Steps>
    </ThemeProvider>,
  );

  expect(tree).toMatchSnapshot();
});

it('正确渲染--垂直方向步进器', () => {
  const tree = renderer.create(
    <ThemeProvider theme={defaultTheme}>
      <Steps current={0} direction="vertical">
        <Step title="第一步" />
        <Step title="第二步" />
        <Step title="第三步" />
      </Steps>
    </ThemeProvider>,
  );

  expect(tree).toMatchSnapshot();
});

it('正确渲染--垂直方向步进器且标签垂直布局', () => {
  const tree = renderer.create(
    <ThemeProvider theme={defaultTheme}>
      <Steps current={0} direction="vertical" labelPlacement="vertical">
        <Step title="第一步" />
        <Step title="第二步" />
        <Step title="第三步" />
      </Steps>
    </ThemeProvider>,
  );

  expect(tree).toMatchSnapshot();
});

describe('classname', () => {
  it('基本使用', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={defaultTheme}>
        <Steps current={0} data-testid="Steps">
          <Step title="第一步" />
          <Step title="第二步" />
          <Step title="第三步" />
        </Steps>
      </ThemeProvider>,
    );

    expect(getByTestId('Steps')).toHaveClass('sinoui-steps');
  });

  it('垂直布局', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={defaultTheme}>
        <Steps current={0} data-testid="Steps" direction="vertical">
          <Step title="第一步" />
          <Step title="第二步" />
          <Step title="第三步" />
        </Steps>
      </ThemeProvider>,
    );

    expect(getByTestId('Steps')).toHaveClass('sinoui-steps--vertical');
  });
});

it('指定onChange属性时，点击单个步进器，onchange被调用', () => {
  const onChange = jest.fn();
  const { container } = render(
    <ThemeProvider theme={defaultTheme}>
      <Steps current={0} data-testid="Steps" onChange={onChange}>
        <Step title="第一步" />
        <Step title="第二步" />
        <Step title="第三步" />
      </Steps>
    </ThemeProvider>,
  );
  const firstStep = container.querySelectorAll('.sinoui-step')[0];

  fireEvent.click(firstStep);

  expect(onChange).toHaveBeenCalled();
});
