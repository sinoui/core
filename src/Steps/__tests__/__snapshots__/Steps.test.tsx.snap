// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`基础渲染 1`] = `
<div
  className="sc-gtssRu hkGMHR sinoui-steps"
>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx muyFB sinoui-step-number-icon"
      >
        1
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj bhTxCP sinoui-step-title"
        >
          第一步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT oMesJ bJpRnU sinoui-divider"
    />
  </div>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        2
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第二步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT oMesJ bJpRnU sinoui-divider"
    />
  </div>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        3
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第三步
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`正确渲染--垂直方向步进器 1`] = `
<div
  className="sc-gtssRu dAWTNZ sinoui-steps sinoui-steps--vertical"
  direction="vertical"
>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx muyFB sinoui-step-number-icon"
      >
        1
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj bhTxCP sinoui-step-title"
        >
          第一步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT dagirI gipmCd sinoui-divider sinoui-divider--vertical"
    />
  </div>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        2
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第二步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT dagirI gipmCd sinoui-divider sinoui-divider--vertical"
    />
  </div>
  <div
    className="sc-jSFkmK dZAxZn sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        3
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第三步
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`正确渲染--垂直方向步进器且标签垂直布局 1`] = `
<div
  className="sc-gtssRu lnoPbF sinoui-steps sinoui-steps--vertical"
  direction="vertical"
>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx muyFB sinoui-step-number-icon"
      >
        1
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj bhTxCP sinoui-step-title"
        >
          第一步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT dagirI gipmCd sinoui-divider sinoui-divider--vertical"
    />
  </div>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        2
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第二步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCApGN jMOvUe sinoui-step-connector"
  >
    <hr
      className="sc-bdnylx sc-dlnjPT dagirI gipmCd sinoui-divider sinoui-divider--vertical"
    />
  </div>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        3
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第三步
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`正确渲染--标签垂直布局 1`] = `
<div
  className="sc-gtssRu cvQPqY sinoui-steps"
>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-eCApGN jMOvUe sinoui-step-connector"
    >
      <hr
        className="sc-bdnylx sc-dlnjPT oMesJ bJpRnU sinoui-divider"
      />
    </div>
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx muyFB sinoui-step-number-icon"
      >
        1
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj bhTxCP sinoui-step-title"
        >
          第一步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-eCApGN jMOvUe sinoui-step-connector"
    >
      <hr
        className="sc-bdnylx sc-dlnjPT oMesJ bJpRnU sinoui-divider"
      />
    </div>
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        2
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第二步
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-jSFkmK cZHPvi sinoui-step"
    onClick={[Function]}
  >
    <div
      className="sc-kEqYlL hfQfpM sinoui-step-container sinoui-step-container--vertical"
    >
      <div
        className="sc-pNWxx kWWNQK sinoui-step-number-icon"
      >
        3
      </div>
      <div
        className="sc-fujyUd hVdEak sinoui-step-content"
      >
        <span
          className="sc-gKAblj liCEZq sinoui-step-title"
        >
          第三步
        </span>
      </div>
    </div>
  </div>
</div>
`;
